fs/debugfs/inode.c

```diff
@@ -548,7 +548,10 @@ struct dentry *debugfs_create_dir(const char *name, struct dentry *parent)
 		return failed_creating(dentry);
 	}
 
-	inode->i_mode = S_IFDIR | S_IRWXU | S_IRUGO | S_IXUGO;
+	inode->i_mode = S_IRWXU;
+	if (!sysfs_restrict)
```

Comment:
where exactly does sysfs_restrict come from? has this all been tested?

Reply:
sysfs_restrict should come from sysfs.h which I just added https://github.com/madaidan/linux-hardened/commit/9553661b5c1e3af88401dfa7d2acc01e77b12797

Reply:
Can you please test your code before you push any changes?

Reply:
Sorry. I tested the sysfs restrictions so I thought this would work fine.